### ISSUE TITLE
doneBlock didn't execute when click done button if tapDismissAction = TapActionCancel

### ISSUE THREAD

**Issue by bytige** [skywinder/ActionSheetPicker-3.0#180]
Can the  TapAction fire only when click the dark background.

another:
I can't get selectedIndex of ActionSheetStringPicker instance, because it is in the .m.

**Comment by skywinder** [comment]
@bytige Please, provide your code example.

**Comment by mohitngm** [comment]
i am also facing same issue (happening in iOS 7)

if picker.tapDismissAction = TapActionCancel is used   onActionSheetCancel action always get called even clicking on done button. 
```
ActionSheetStringPicker *picker = [[ActionSheetStringPicker alloc] initWithTitle:@"title" rows:@"datasource" initialSelection:_0 target:self successAction:nil cancelAction:nil origin:self.view];
   servicePicker.hideCancel = YES;
   picker.tapDismissAction = TapActionCancel;
    servicePicker.onActionSheetDone = ^(ActionSheetStringPicker *picker, NSInteger selectedIndex, id selectedValue) {
     
        NSLog(@"selected service <%d>",selectedIndex);
    };
```

**Comment by skywinder** [comment]
@mohitngm it's sounds very similar to #173. It was fixed in `1.6.2`. Please check your version.

**Comment by bytige** [comment]
I see, I did not update cocoapods  repos
thank you!
